components/tidb_query/src/codec/mysql/decimal.rs

```diff
@@ -1710,21 +1720,17 @@ enable_conv_for_int!(usize, u64);
 enable_conv_for_int!(isize, i64);
 
 impl ConvertTo<f64> for Decimal {
+    /// This function should not return err,
+    /// if it return err, then the err because of bug.
+    ///
+    /// Port from TiDB's MyDecimal::ToFloat64.
```

Comment:
However, the parse function is very likely to has some small bug, but such bug is not fatal, so if we `unwrap` here, the tikv will panic because of such small bug, which is very bad for our user

Reply:
Should we construct a mechanism to handle such error which because of some bug, or we will loss  info of such some bug when meet it.

Reply:
Please consider removing this comment. If it fails to parse, no matter what reason `tikv` should not fail because of some invalid data format. We need to return an error.

Reply:
```rust
impl ConvertTo<f64> for Decimal {
    /// This function should not return err,
    /// if it return err, then the err because of bug.
    ///
    /// Port from TiDB's MyDecimal::ToFloat64.
    #[inline]
    fn convert(&self, _: &mut EvalContext) -> Result<f64> {
        let val = self.to_string().parse()?;
        Ok(val)
    }
}
```

the string is from `decimal::to_string`, so it should not be invalid string

Reply:
I think for any "should" they deserve an assertion. You shall either use assertion for your "should", or change comment to "may".

Reply:
Ok, I will add `debug_assert!` in such place, and I think the doc is necessary